src/sentry/plugins/sentry_mail/activity/release.py

```diff
@@ -100,20 +100,38 @@ def get_participants(self):
         if not self.email_list:
             return {}
 
-        # identify members which have been seen in the commit log and have
-        # verified the matching email address
-        return {
-            user: GroupSubscriptionReason.committed
+        # collect all users with verified emails on a team in the related projects,
+        # and couple them with the the users' setting value for deploy-emails
+        users_with_options = [
+            (user, UserOption.objects.get_value(
+                user=user,
+                organization=self.organization,
+                key='deploy-emails',
+                default=UserOptionValue.committed_deploys_only,
+            ))
             for user in User.objects.filter(
-                in_iexact('emails__email', self.email_list),
                 emails__is_verified=True,
                 sentry_orgmember_set__teams=Team.objects.filter(
                     id__in=[p.team_id for p in self.projects]
                 ),
                 is_active=True,
             ).distinct()
+        ]
+
+        participants = {
+            # assign reason they were included:
+            user: GroupSubscriptionReason.committed if
+            option == UserOptionValue.committed_deploys_only else GroupSubscriptionReason.deploy_setting
+
+            for user, option in users_with_options
+
+            # filter down to members which have been seen in the commit log:
+            if option == UserOptionValue.committed_deploys_only and user.email in self.email_list or
+            option == UserOptionValue.all_deploys  # or who opt into all deploy emails:
```

Comment:
the spacing / comments definitely help, but it might be better to just do this not as a dictionary comprehension?